mmaction/datasets/repeat_aug_dataset.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from copy import deepcopy
+from typing import Any, Callable, List, Optional, Sequence, Union
+
+import numpy as np
+from mmengine.dataset import COLLATE_FUNCTIONS, pseudo_collate
+
+from mmaction.registry import DATASETS
+from mmaction.utils import ConfigType
+from .video_dataset import VideoDataset
+
+
+def get_type(transform: Union[dict, Callable]) -> str:
+    """get the type of the transform."""
+    if isinstance(transform, dict) and 'type' in transform:
+        return transform['type']
+    elif callable(transform):
+        return transform.__repr__().split('(')[0]
+    else:
+        raise TypeError
+
+
+@DATASETS.register_module()
+class RepeatAugDataset(VideoDataset):
+    """Video dataset for action recognition.
+
+    The dataset loads raw videos and apply specified transforms to return a
+    dict containing the frame tensors and other information.
+
+    The ann_file is a text file with multiple lines, and each line indicates
+    a sample video with the filepath and label, which are split with a
+    whitespace. Example of a annotation file:
+
+    .. code-block:: txt
+
+        some/path/000.mp4 1
+        some/path/001.mp4 1
+        some/path/002.mp4 2
+        some/path/003.mp4 2
+        some/path/004.mp4 3
+        some/path/005.mp4 3
+
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        pipeline (List[Union[dict, ConfigDict, Callable]]): A sequence of
+            data transforms.
+        data_prefix (dict or ConfigDict): Path to a directory where videos
+            are held. Defaults to ``dict(video='')``.
+        multi_class (bool): Determines whether the dataset is a multi-class
+            dataset. Defaults to False.
+        num_classes (int, optional): Number of classes of the dataset, used in
+            multi-class datasets. Defaults to None.
+        start_index (int): Specify a start index for frames in consideration of
+            different filename format. However, when taking videos as input,
+            it should be set to 0, since frames loaded from videos count
+            from 0. Defaults to 0.
+        modality (str): Modality of data. Support ``RGB``, ``Flow``.
+            Defaults to ``RGB``.
+        test_mode (bool): Store True when building test or validation dataset.
+            Defaults to False.
+    """
+
+    def __init__(self,
+                 ann_file: str,
+                 pipeline: List[Union[dict, Callable]],
+                 data_prefix: ConfigType = dict(video=''),
+                 num_repeats: int = 4,
+                 multi_class: bool = False,
+                 num_classes: Optional[int] = None,
+                 start_index: int = 0,
+                 modality: str = 'RGB',
+                 **kwargs) -> None:
+
+        flag = get_type(pipeline[0]) == 'DecordInit' and \
```

Comment:
We should avoid using ambiguous variable names like `flag`. What about `backend_supported`?